build(android): update v8 download logic

sgtcoolguy opened the pull request:
**Description:**
This is a work in progress to see if there's some build caching we can use to speed up CI builds in particular.

- [X] use ssri integrity hash from our build utils, should allow ci server to cache downloaded archive
- [X] use folder-hash npm package to cache extracted copies of v8/modules under the tmp dir. Then copy the extracted contents to ultimate destinations

So the downside here is that we download/extract to some tmp location so we could eat up disk space with the extracted copies (hence why I do so in the tmp dir). Additionally, I do not avoid the downloading of the archives if the extracted copy already exists (or the extraction if the final destination dir exists). So I guess we could be doing unnecessary steps of downloading/extracting when you already have the final extracted contents in the `dist` dir but don't have the intermediate .tar.bz2/.zip or the tmp dir copy of the extracted contents.

sgtcoolguy committed:
build(android): update v8 download logic

use ssri integrity hash from our build utils, should allow ci server to cache downloaded archive

build commented:
<!--
  0 failure: 
  2 warning:  There is no linke..., :mag: Can't find ...
  2 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :mag: Can't find junit reports at `./junit.*.xml`, skipping generating JUnit Report.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:thumbsup: Hey!, You deleted more code than you added. That's awesome!</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3c0d1924b8543857eadb1e6b7370c28dac598c99
</p>

sgtcoolguy committed:
build(android): set integrity hash for v8

sgtcoolguy committed:
chore(package): add folder-hash 3.3.0

sgtcoolguy committed:
build: cache unzipped copies of native modules

hopefully speeds up builds on CI servers that get bogged down with unzipping modules
we extract once to tmp dir and record hash for contents and on later builds
we try to use pre-extracted cache if it exists

sgtcoolguy committed:
build: ensure modules dir is created before module extraction

sgtcoolguy committed:
build: attempt to avoid mkdir collisions for native modules

sgtcoolguy committed:
build: handle '/' char in integrity hash for caching

sgtcoolguy committed:
build: cache extraction of V8 tar.bz2 file

sgtcoolguy committed:
Merge branch 'master' into ci-cache

sgtcoolguy committed:
Merge branch 'master' into ci-cache

sgtcoolguy committed:
Merge branch 'master' into ci-cache
